gprofiler/application_separators.py

```diff
@@ -0,0 +1,232 @@
+        wsgi = get_cli_arg_value(process.cmdline(), "-w")
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:3]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_cli_arg_value(process.cmdline(), "-A") or get_cli_arg_value(
+            process.cmdline(), "--app", check_for_equals_arg=True
+        )
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "PySpark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return (
+            len(process.cmdline()) >= 3
+            and "python" in process.cmdline()
+            and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+        )
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "PySpark"
+
+
+class PythonModuleApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "Python"
+
+    def is_supported(self, process: Process) -> bool:
+        if not _is_python_bin(process.cmdline()[0]):
+            return False
+
+        return (len(process.cmdline()) >= 3 and process.cmdline()[1] == "-m") or (
+            len(process.cmdline()) >= 2 and process.cmdline()[1].endswith(".py")
+        )
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        module_arg = get_cli_arg_value(process.cmdline(), "-m")
+        if module_arg is not None:
+            return module_arg
+
+        module_filename = process.cmdline()[1][:-3]  # Strip the ".py" (checked before)
+        return append_python_module_to_proc_wd(process, module_filename)
+
+
+class JavaJarApplicationSeparator(ApplicationSeparator):
+    def is_supported(self, process: Process) -> bool:
+        return "java" in os.path.basename(process.cmdline()[0]) and "-jar" in process.cmdline()
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return get_cli_arg_value(process.cmdline(), "-jar")
+
+    @property
+    def app_prefix(self) -> str:
```

Comment:
`app_prefix` is the first function in your other classes, reorder

Reply:
don't relevant anymore